app/lib/message_list/message_composer.dart

```diff
@@ -354,6 +369,35 @@ class _MessageComposerState extends State<MessageComposer>
       );
     });
     _emojiAutocomplete.handleTextChanged();
+    _scheduleActionButtonSizeUpdate();
+  }
+
+  void _scheduleActionButtonSizeUpdate() {
+    if (_actionButtonSizeUpdateScheduled) {
+      return;
+    }
+    _actionButtonSizeUpdateScheduled = true;
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      _actionButtonSizeUpdateScheduled = false;
+      if (!mounted) {
+        return;
+      }
+      _updateActionButtonSize();
+    });
+  }
+
+  void _updateActionButtonSize() {
+    final newHeight = _inputFieldKey.currentContext?.size?.height;
+    if (newHeight == null || newHeight <= 0) {
+      return;
+    }
+    final targetHeight = newHeight.clamp(0.0, _maxActionButtonSize).toDouble();
+    if ((_actionButtonSize - targetHeight).abs() < 0.5) {
+      return;
+    }
+    setState(() {
+      _actionButtonSize = targetHeight;
+    });
   }
```

Comment:
Instead of calculating the size of the button from the height of the text field, shouldn't we rather set the height of the text field? The problem is that text field expands its height when multiple lines are input, and then the button still grows slightly, even though its height is constrained.

The formula for a text field with `dense = true` is:

TotalHeight = (FontSize × LineHeight) + (VerticalPadding × 2)

Reply:
Settings the height of the action button with intrinsic height also works, but only for the text field containing a single line. When it expand the button also expands then.